rosidl_parser/rosidl_parser/__init__.py

```diff
@@ -705,3 +722,76 @@ def parse_service_string(pkg_name, srv_name, message_string):
         pkg_name, srv_name + SERVICE_RESPONSE_MESSAGE_SUFFIX, response_message_string)
 
     return ServiceSpecification(pkg_name, srv_name, request_message, response_message)
+
+def parse_action_file(pkg_name, interface_filename):
+    basename = os.path.basename(interface_filename)
+    action_name = os.path.splitext(basename)[0]
+    with open(interface_filename, 'r') as h:
+        return parse_action_string(pkg_name, action_name, h.read())
+
+
+def parse_action_string(pkg_name, action_name, action_string):
+    action_blocks = action_string.split(ACTION_REQUEST_RESPONSE_SEPARATOR)
+    if len(action_blocks) != 3:
+        raise InvalidSpecification(
+            "Number of '%s' separators nonconformant with action definition" %
+            ACTION_REQUEST_RESPONSE_SEPARATOR)
+
+    goal_service_string, result_service_string, feedback_message_string = action_blocks
+
+    services = []
+    #----------------------------------------------------------------------------------------------
+    ## Send goal
+    implicit_input = ["uint8[16] uuid"]
+    request_message_string = goal_service_string + '\n' + '\n'.join(implicit_input)
+    request_message = parse_message_string(
+        pkg_name,
+        action_name + ACTION_GOAL_MESSAGE_SUFFIX + SERVICE_REQUEST_MESSAGE_SUFFIX,
+        request_message_string)
+
+    implicit_output = ["bool accepted", "builtin_interfaces/Time stamp"]
```

Comment:
@jacobperron @hidmic Will `rcl_actions` associate the service response by mapping goal ID to the request sequence number, or by having the higher level client library get the goal ID out of the message? If the former then these fields should be OK, if the latter then the stamp field should be `action_msgs/GoalInfo` so it includes a goal id too.

Reply:
I think that'd be currently the case (using goal IDs I mean).